code/modules/farming/soil.dm

```diff
@@ -844,6 +844,9 @@
 	var/turf/location = loc
 	if(!plant.can_grow_underground && !location.can_see_sky())
 		return
+	var/obj/structure/soil/box/box = locate(/obj/structure/soil/box) in location
+	if(box && !plant.can_grow_boxed)
+		return
```

Comment:
```suggestion
	var/obj/structure/soil/box/box = locate(/obj/structure/soil/box) in location
	if(box && !plant.can_grow_boxed)
		return
```
override process_plant_nutrition() on soil/box to do this if(plant.can_grow_boxed) check then call parent